src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -287,10 +363,32 @@ func (s *peersSource) fetchBootstrapBlocksFromPeers(
 				continue
 			}
 
-			// If not waiting to flush, add straight away to bootstrap result
-			lock.Lock()
-			bootstrapResult.Add(shard, shardResult, xtime.Ranges{})
-			lock.Unlock()
+			// If not waiting to flush, add straight away to bootstrap result.
+			for _, elem := range shardResult.AllSeries().Iter() {
+				entry := elem.Value()
+
+				tagsIter.Reset(entry.Tags)
+				ref, err := accumulator.CheckoutSeries(entry.ID, tagsIter)
+				if err != nil {
+					unfulfill(currRange)
+					s.log.Error("could not checkout series", zap.Error(err))
+					continue
+				}
+
+				for t, block := range entry.Blocks.AllBlocks() {
+					if err := ref.Series.LoadBlock(block, series.WarmWrite); err != nil {
+						unfulfill(currRange)
+						s.log.Error("could not load series block", zap.Error(err))
+					}
+					entry.Blocks.RemoveBlockAt(t.ToTime())
```

Comment:
Could you clear up why this needs to remove blocks here but Why does this removes blocks here, but not in other bootstrappers?

Also, is it safe to remove it from AllBlocks while iterating like this?

Reply:
Also this will remove it whether or not there's an error, is this intentional?

Reply:
I'll remove the removing of blocks, it's not really necessary (was a previous over optimization that we can't really make anymore to return blocks to the pool).